packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -80,31 +110,118 @@ export default class RoutesHandler {
+            res.status(HttpStatusCodes.CREATED).send(message);
+            this.logger.debug(`XML Fragment with name "${fragmentName}" was created`);
+        } catch (e) {
+            const sanitizedMsg = sanitize(e.message);
+            this.logger.error(sanitizedMsg);
+            res.status(HttpStatusCodes.INTERNAL_SERVER_ERROR).send(sanitizedMsg);
+            next(e);
+        }
+    };
+
+    /**
+     * Handler for reading all controller extension files from the workspace.
+     *
+     * @param _ Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleReadAllControllers = async (_: Request, res: Response, next: NextFunction) => {
+        try {
+            const files = await this.readAllFilesByGlob('/**/changes/coding/*.js');
+
+            const fileNames = files.map((f) => ({
+                controllerName: f.getName()
+            }));
+
+            this.sendFilesResponse(res, {
+                controllers: fileNames,
+                message: `${fileNames.length} controllers found in the project workspace.`
+            });
+            this.logger.debug(`Read controllers ${JSON.stringify(fileNames)}`);
+        } catch (e) {
+            this.handleErrorMessage(res, next, e);
+        }
+    };
+
+    /**
+     * Handler for writing a controller extension file to the workspace.
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleWriteControllerExt = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const data = req.body as WriteControllerBody;
+
+            const controllerExtName = sanitize(data.controllerName);
+
+            const sourcePath = this.util.getProject().getSourcePath();
+            const rootPath = this.util.getProject().getRootPath();
+            const projectFolderName = rootPath.split(/[\\/]/).pop();
+
+            if (!controllerExtName) {
+                res.status(HttpStatusCodes.BAD_REQUEST).send('Controller extension name was not provided!');
+                this.logger.debug('Bad request. Controller extension name was not provided!');
+                return;
+            }
+
+            const fullPath = path.join(sourcePath, FolderNames.Changes, FolderNames.Coding);
+            const filePath = path.join(fullPath, `${controllerExtName}.js`);
+
+            if (!fs.existsSync(fullPath)) {
+                fs.mkdirSync(fullPath, { recursive: true });
             }
+
+            if (fs.existsSync(filePath)) {
+                res.status(HttpStatusCodes.CONFLICT).send(
+                    `Controller extension with name "${controllerExtName}" already exists`
+                );
+                this.logger.debug(`Controller extension with name "${controllerExtName}" already exists`);
+                return;
+            }
+
+            const controllerExtPath = `${projectFolderName}.${controllerExtName}`;
```

Comment:
I am not sure I understand this concatenation here. Why would you use the project folder name as part of the name of the controller? Shouldn't that be the application id taken from the manifest?

Reply:
Changed in: https://github.com/SAP/open-ux-tools/pull/1275/commits/83cc9215a49b550916cfdb9fb7a5522aa471011b

Reply:
Solves the issue. It is not exactly how I would have it implemented because you know the ID already in your middleware so there is no need to have the client send it but it is an acceptable solution.